tests/bluetooth/bsim_bt/bsim_test_audio/src/has_client_test.c

```diff
@@ -0,0 +1,279 @@
+	uint8_t id_prev, id_curr;
+
+	static struct bt_gatt_exchange_params mtu_params = {
+		.func = mtu_cb,
+	};
+
+	err = bt_enable(bt_ready);
+	if (err < 0) {
+		FAIL("Bluetooth discover failed (err %d)\n", err);
+		return;
+	}
+
+	printk("Bluetooth initialized\n");
+
+	WAIT_FOR(g_bt_init);
+
+	err = bt_has_client_cb_register(&has_cb);
+	if (err < 0) {
+		FAIL("Callback registration (err %d)\n", err);
+		return;
+	}
+
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, device_found);
+	if (err < 0) {
+		FAIL("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning succesfuly started\n");
+
+	WAIT_FOR(g_is_connected);
+
+	err = bt_gatt_exchange_mtu(g_conn, &mtu_params);
+	if (err < 0) {
+		FAIL("Failed to exchange MTU (err %d)\n", err);
+		return;
+	}
+
+	err = bt_has_discover(g_conn);
+	if (err < 0) {
+		FAIL("Failed to discover HAS (err %d)\n", err);
+		return;
+	}
+
+	WAIT_FOR(g_has_discovered);
+
+	err = bt_has_preset_active_get(g_has);
+	if (err < 0) {
+		FAIL("Failed to get active preset id (err %d)\n", err);
+		return;
+	}
+
+	WAIT_FOR(g_active_preset_notified);
+
+	printk("Got active preset id %d\n", g_active_preset_id);
+
+	id_prev = g_active_preset_id;
+
+	/* Reset the notification indicator */
+	g_active_preset_notified = false;
+
+	err = bt_has_preset_active_set_next(g_has);
+	if (err < 0) {
+		FAIL("Failed to set next (err %d)\n", err);
+		return;
+	}
+
+	WAIT_FOR(g_active_preset_notified);
+
+	id_curr = g_active_preset_id;
+
+	if (id_curr == id_prev) {
+		FAIL("Preset not changed");
+		return;
+	}
+
+	printk("Got active preset id %d\n", g_active_preset_id);
+
+	id_prev = g_active_preset_id;
+
+	/* Reset the notification indicator */
+	g_active_preset_notified = false;
+
+	err = bt_has_preset_active_set_next(g_has);
+	if (err < 0) {
+		FAIL("Failed to set next (err %d)\n", err);
+		return;
+	}
+
+	WAIT_FOR(g_active_preset_notified);
+
+	id_curr = g_active_preset_id;
+
+	if (id_curr == id_prev) {
+		FAIL("Preset not changed");
+		return;
+	}
+
+	printk("Got active preset id %d\n", g_active_preset_id);
+
+	/* Reset the notification indicator */
+	g_active_preset_notified = false;
+
+	err = bt_has_preset_active_set_prev(g_has);
+	if (err < 0) {
+		FAIL("Failed to set prev (err %d)\n", err);
+		return;
+	}
+
+	WAIT_FOR(g_active_preset_notified);
+
+	id_curr = g_active_preset_id;
+
+	if (id_curr != id_prev) {
+		FAIL("Failed to set to previous %d != %d\n", id_curr, id_prev);
+		return;
+	}
+
+	printk("Got active preset id %d\n", g_active_preset_id);
+
+	err = bt_has_preset_active_clear(g_has);
+	if (err == 0) {
+		FAIL("Client cannot perform Active Preset clear operation\n");
+		return;
+	}
+
+	err = bt_has_preset_read_multiple(g_has, 0x01, 0xff);
+	if (err < 0) {
+		FAIL("Failed to read all presets (err %d)\n", err);
+		return;
+	}
+
+	WAIT_FOR(g_preset_read);
+
+	err = bt_has_preset_name_set(g_has, id_curr, "Custom");
+	if (err < 0) {
+		FAIL("Failed to set name (err %d)\n", err);
+		return;
+	}
+
+	k_sleep(K_MSEC(100));
```

Comment:
Why are we waiting 100ms here?

Reply:
Waiting for notification, fixed with condition WAIT_FOR